server/src/test/java/io/envoyproxy/controlplane/server/serializer/CachedProtoResourcesSerializerTest.java

```diff
@@ -0,0 +1,35 @@
+package io.envoyproxy.controlplane.server.serializer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.Lists;
+import com.google.protobuf.Any;
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment;
+import java.util.Collection;
+import java.util.List;
+import org.junit.Test;
+
+public class CachedProtoResourcesSerializerTest {
```

Comment:
Yeah, was thinking about this, but since `System.gc()` does not guarantee GC it's hard to test it.

Reply:
This is part of why I'm not a huge fan of these gc based retention policies, since we have no way of verifying that changes made in other PRs won't break the cache performance. That said they do elegantly solve this problem so I'm not completely opposed to them